security/sshdo/Portfile

```
@@ -0,0 +1,67 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           makefile 1.0
+
+github.setup        raforg sshdo 1.1.1 v
+revision            0
+
+categories          security
+license             GPL-2+
+maintainers         {raf.org:raf @macportsraf} openmaintainer
+description         controls which commands may be executed via incoming ssh
+long_description    {*}${description}. \
+                    \n\nsshdo provides an easily configurable way of controlling which \
+                    commands may be executed via incoming ssh connections by specific \
+                    users and specific keys. Training mode allows a key to be used for \
+                    any command, while logging them. Then sshdo can learn from the logs \
+                    to know what commands to allow. It can also unlearn commands that \
+                    are no longer in use. This mitigates against private ssh key compromise \
+                    with very little effort.
+# Any of Python 2.6, 2.7, or 3.3+ will do.
+# When updating the python3 version, update the corresponding patch below as well.
+# Update the python3 version every year (or until there's a meta port:python3).
+depends_lib-append  port:python311
+
+homepage            https://raf.org/sshdo/
+github.tarball_from releases
+checksums           rmd160 3071b9aa06452437f9da970f0576c94ea27d3c08 \
+                    sha256 9b22e14fec786b692e8d6431a81e9b17ad8f6e62ae5d6ef9e9c643690bf33f90 \
+                    size 79485
+
+# Replace "python3" with the specific version depended-on above.
+# This means the user doesn't have to select a default python3 version.
+patchfiles          macports.patch
+
+# Note: Not a GNU autoconf configure script
+use_configure       yes
+configure.args      --macports-prefix=${prefix}
+build {}
+
+test.run            yes
+
+destroot.keepdirs   ${destroot}${prefix}/etc/sshdoers.d
+
+notes {
+    sshdo relies on syslog messages to learn and unlearn policy.
+
+    On recent macOS systems with the unified logging system,
+    syslog messages don't look the same as elsewhere, and they no
+    longer appear in /var/log/system.log where sshdo expects them,
+    so you'll need to use the log(1) command to feed log messages
+    into sshdo via its standard input (stdin). See man sshdo.
+
+    Use the following command as the basis for your log queries:
+
+        log show --style syslog --info --predicate '
+            process == "Python" and
+            composedMessage contains "type=" and
+            composedMessage contains "user=" and
+            composedMessage contains "command="
+        ' | sed -E -e 's/Python/sshdo/' -e 's/: \([^)]+\)/:/'
+
+    You will also need to incorporate the appropriate time range
+    into the search criteria (e.g., --last 7d). See man log.
+}
+
```

Instead of adding newlines into paragraphs manually, please allow MacPorts to add them automatically based on the user's terminal width:

```suggestion
notes {
    sshdo relies on syslog messages to learn and unlearn policy.

    On recent macOS systems with the unified logging system,\
    syslog messages don't look the same as elsewhere, and they no\
    longer appear in /var/log/system.log where sshdo expects them,\
    so you'll need to use the log(1) command to feed log messages\
    into sshdo via its standard input (stdin). See man sshdo.

    Use the following command as the basis for your log queries:

        log show --style syslog --info --predicate '
            process == "Python" and
            composedMessage contains "type=" and
            composedMessage contains "user=" and
            composedMessage contains "command="
        ' | sed -E -e 's/Python/sshdo/' -e 's/: \([^)]+\)/:/'

    You will also need to incorporate the appropriate time range\
    into the search criteria (e.g., --last 7d). See man log.
}

```

---

Thanks. It's done.